File: Frontend/Private/gco-private/src/app/model/core/temp-score.ts
import { ScoreType } from './score-type'
import { Apparatus } from './apparatus'
import { ContestantScoring } from './contestant-scoring'
import { JudgeBasicInfo } from '../dto/judge-basic-info'

export interface TempScore{
    id? : string;
    type : ScoreType;
    apparatus : Apparatus;
    value : number;
    contestant : ContestantScoring;
    competitionId: string;
    judge : JudgeBasicInfo;
}


export interface TempScoreTable {
    displayedColumns: string[];
    dataSource: TempScore[];
    selectedRow: TempScore | null;
}

File: Backend/scoring_service/core/domain/temp_score.go
package domain

import "github.com/google/uuid"

type TempScore struct {
	ID            uuid.UUID
	Type          ScoreType
	Apparatus     Apparatus
	Value         float32
	ContestantID  uuid.UUID
	Contestant    Contestant
	CompetitionID uuid.UUID
	Competition   Competition
	JudgeID       uuid.UUID
	Judge         Judge
}
